File: Nexo/Models/PlatformUser.swift
//
//  PlatformUser.swift
//  Nexo
//
//  Created by Atzmi, Dan on 25/04/2026.
//

import Foundation

struct PlatformUser: Identifiable {
    let id: String // Firebase UID
    var firstName: String
    var lastName: String
    var email: String
    var role: PlatformRole

    var fullName: String {
        "\(firstName) \(lastName)".trimmingCharacters(in: .whitespaces)
    }

    var displayName: String {
        fullName.isEmpty ? email : fullName
    }
}

File: android/app/src/main/java/com/nexo/app/domain/model/PlatformUser.kt
package com.nexo.app.domain.model

/**
 * A platform-wide user record, as seen from the Platform Admin Dashboard's
 * Users tab — mirrors iOS's `PlatformUser`. `id` is the Firebase Auth UID,
 * matching `users/{uid}` — see FIRESTORE_SCHEMA.md.
 */
data class PlatformUser(
    val id: String,
    val firstName: String,
    val lastName: String,
    val email: String,
    val role: PlatformRole = PlatformRole.USER,
    val profilePicBase64: String? = null
) {
    val fullName: String get() = "$firstName $lastName".trim()
    val displayName: String get() = fullName.ifEmpty { email }
}
